### PR TITLE
Update korean translation to 84af306

### PR DESCRIPTION
**Pull request by marocchino**

### CODE REVIEW

File: docs/docs/02-displaying-data.ko-KR.md [hunk 1]
```diff
@@ -91,12 +89,11 @@ JSX를 이용하면:
...
 우리는 이것이 React 앱들을 만들기 쉽게 하고, 디자이너들이 이 문법을 더 선호하는 것을 발견했습니다, 하지만 모든 사람은 그들만의 선호하는 워크플로우가 있기 마련이므로, **JSX는 React를 사용하기 위해 필수적이지는 않습니다.**
 
-JSX는 매우 작은 언어입니다. 그것을 배우고 싶다면, [JSX 깊게 살펴보기](/react/docs/jsx-in-depth-ko-KR.html). 를 살펴 보시기 바랍니다. 또는, [우리의 온라인 JSX 컴파일러](/react/jsx-compiler.html)를 통해 문법이 변환되는 것을 살펴 보시기 바랍니다.
+JSX는 매우 작은 언어입니다. 그것을 배우고 싶다면, [JSX 깊게 살펴보기](/react/docs/jsx-in-depth-ko-KR.html)를 살펴 보시기 바랍니다. 또는, [바벨 REPL](https://babeljs.io/repl/)를 통해 문법이 변환되는 것을 살펴 보시기 바랍니다.
 
 JSX는 HTML과 비슷하지만, 완전히 똑같지는 않습니다. [JSX의 실수하기 쉬운 부분들](/react/docs/jsx-gotchas-ko-KR.html)에 중요한 차이점들에 대해 설명되어 있습니다.
 
-JSX를 사용하기 시작하기 위한 가장 쉬운 방법은 브라우저에서 작동하는 `JSXTransformer`를 사용하는 것입니다. 우리는 이것을 프로덕션에서는 사용하지 않기를 강하게 권장하는 바입니다. 당신은 우리의 커맨드 라인 [react-tools](https://www.npmjs.com/package/react-tools) 패키지를 이용하여 미리 컴파일(precompile)해 사용할 수 있습니다.
-
+[바벨에서 JSX를 시작하는 여러 방법을 제공합니다](http://babeljs.io/docs/setup/) 여기에는 커맨드 라인 툴에서 루비 온 레일스 연동까지 포함됩니다. 가장 편한 툴을 사용하세요.
```
Comment (marocchino) [inline docs/docs/02-displaying-data.ko-KR.md]: 커맨드 라인 툴에서 루비 온 레일스 연동까지 포함됩니다. -> 커맨드 라인 툴부터 루비 온 레일스 연동까지 다양한 방법이 있습니다.

File: docs/docs/05-reusable-components.ko-KR.md [hunk 2]
```diff
@@ -230,3 +228,30 @@ Counter.defaultProps = { initialCount: 0 };
...
+ReactDOM.render(<HelloMessage name="Sebastian" />, mountNode);
+```
+
+이 단순화된 컴포넌트 API는 prop의 순수 함수인 컴포넌트를 나타냅니다. 이 컴포넌트는 내부 상태가 없어야 하고, 내부 인스턴스가 없어야 하고, 컴포넌트 생명주기 메소드가 없어야 합니다. 아무런 준비 과정없이 입력에 대한 순수한 기능적 변환이어야 합니다.
+
+> 주의:
+>
+> 상태를 가지지 않는 함수는 내부 인스턴스가 없기 때문에, ref를 상태를 가지지않는 함수에 넣을 수 없습니다. 상태를 가지지 않는 함수는 명령형(imperative) API를 제공하지 않기 때문에 일반적으로 이것은 문제가 되지 않습니다. 명령형 API없이 인스턴스에 할 수 있는 것이 많지 않기도 하죠. 하지만 상태를 가지지 않는 컴포넌트의 DOM 노드를 검색하길 원한다면, 반드시 상태 기반 컴포넌트(예. ES6 클래스 컴포넌트)로 컴포넌트를 감싸고 상태 기반 래퍼 컴포넌트에 ref를 붙여야 합니다.
+
+이상적으로는, 대부분의 컴포넌트는 상태를 가지지 않는 함수여야 합니다. 왜냐 하면 이런 상태 없는 컴포넌트는 React 코어 안에서 더 빠른 코드 경로를 거치기 때문입니다. 이는 가능한 한 추천하는 패턴입니다.
```
Comment (marocchino) [inline docs/docs/05-reusable-components.ko-KR.md]: 상태 없는 -> 상태를 가지지 않는

File: docs/docs/08.1-more-about-refs.ko-KR.md [hunk 3]
```diff
@@ -1,137 +1,125 @@
...
+// 여기 코드가 위치합니다...
+
+var myComponentInstance = ReactDOM.render(myComponentElement, myContainer);
+myComponentInstance.doSomething();
+```
 
 > 주의:
 >
-> 기억하세요, `render()`에서 반환된 것은 *실제* 자식 인스턴스가 아닙니다. 단지 특정 시점의 컴포넌트의 하위 계층구조에 있는 자식 인스턴스의 *서술*일 뿐입니다.
+> 이는 오직 최상위 레벨에서만 사용되어야 합니다. 컴포넌트의 내부에서는 `props`와 `state`가 자식컴포넌트와 통신하도록 하거나 문자열이나 콜백 어트리뷰트 등 ref를 얻어오는 다른 방법을 사용하도록 하세요.
```
Comment (marocchino) [inline docs/docs/08.1-more-about-refs.ko-KR.md]: 자식컴포넌트와 -> 자식 컴포넌트와

File: docs/docs/08.1-more-about-refs.ko-KR.md [hunk 4]
```diff
@@ -1,137 +1,125 @@
...
+        ref={function(input) {
+          if (input != null) {
+            input.focus();
+          }
+        }} />
     );
-  }
+  },
 ```
+ES6 arrow function을 사용한다면 다음과 같습니다:
```
Comment (marocchino) [inline docs/docs/08.1-more-about-refs.ko-KR.md]: arrow function -> 화살표 함수

File: docs/docs/08.1-more-about-refs.ko-KR.md [hunk 5]
```diff
@@ -1,137 +1,125 @@
...
+ReactDOM.render(
+  <MyComponent />,
+  document.getElementById('example')
+);
 ```
 
-이 예제에서 render 함수는 `<input />` 인스턴스의 서술을 반환했지만, 진짜 인스턴스는 `this.refs.theInput`을 통해 접근할 수 있었습니다. `render`에서 `ref="theInput"`을 가진 컴포넌트가 반환된 동안, `this.refs.theInput`은 적절한 인스턴스에 접근합니다. 이는 `<Typeahead ref="myTypeahead" />`처럼 DOM이 아닌 고수준의 컴포넌트에서도 동작합니다.
+이 예제에서, 텍스트 인풋의 **지원 인스턴스**에 대한 참조를 얻었으며 버튼이 클릭되었을때 `focus()`를 호출했습니다.
 
+합성 컴포넌트에서는 참조는 컴포넌트 클래스의 인스턴스를 가르키기 때문에 클래스에 정의된 어떤 메소드도 호출할 수 있습니다. 컴포넌트의 기저에 있는 DOM 노드에 접근하기 위해서는 [ReactDOM.findDOMNode](/react/docs/top-level-api-ko-KR.html#reactdom.finddomnode)를 "탈출구"로 사용할 수 있습니다. 하지만 이는 캡슐화를 깨며, 대부분의 경우 React 모델을 이용해 더 명확한 방법으로 구조를 짤 수 있기 때문에 추천하지 않습니다.
```
Comment (marocchino) [inline docs/docs/08.1-more-about-refs.ko-KR.md]: 가르키기 -> 가리키기

File: docs/docs/10.1-animation.ko-KR.md [hunk 6]
```diff
@@ -240,11 +239,7 @@ var ImageCarousel = React.createClass({
 </ReactTransitionGroup>
 ```
 
-모든 React가 렌더할 수 있는 DOM 컴포넌트는 사용할 수 있습니다. 하지만 `component`가 DOM 컴포넌트일 필요는 없습니다. React 컴포넌트라면 무엇이든 넣을 수 있습니다. 직접 구현한 컴포넌트여도 됩니다!
-
-> 주의:
->
-> v0.12이전에는, DOM 컴포넌트를 사용할 때, `component` prop은 `React.DOM.*`로 참조할 필요가 있었습니다. 이제 컴포넌트가 단순히 `React.createElement`로 전달되기 때문에, `component` prop은 스트링이어야 합니다. 복합 컴포넌트는 팩토리를 넘겨야 합니다.
+모든 React가 렌더할 수 있는 DOM 컴포넌트는 사용할 수 있습니다. 하지만 `component`가 DOM 컴포넌트일 필요는 없습니다. React 컴포넌트라면 무엇이든 넣을 수 있습니다. 직접 구현한 컴포넌트여도 됩니다! 그냥 `component={List}`를 적으면 컴포넌트는 `this.props.children`로 받을 수 있습니다.
```
Comment (marocchino) [inline docs/docs/10.1-animation.ko-KR.md]: 모든 React가 렌더할 수 있는 DOM 컴포넌트는 -> React가 렌더할 수 있는 DOM 컴포넌트는 전부

File: docs/docs/12-context.ko-KR.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
+---
+id: context
+title: 컨텍스트
+permalink: context-ko-KR.html
+prev: advanced-performance-ko-KR.html
+---
+
+React의 가장 큰 장점 중 하나는 React 컴포넌트를 통해 데이터의 흐름을 추적하기 쉽다는 것입니다. 컴포넌트를 보면 각각의 프로퍼티가 어떻게 전달되었는지 쉽게 파악할 수 있습니다. 
+
+Occasionally, you want to pass data through the component tree without having to pass the props down manually at every level. React's "context" feature lets you do this.
```
Comment (marocchino) [inline docs/docs/12-context.ko-KR.md]: remove it!

File: docs/docs/12-context.ko-KR.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+  render: function() {
+    var children = this.props.messages.map(function(message) {
+      return <Message text={message.text} />;
+    });
+    return <div>{children}</div>;
+  }
+});
+```
+
+`childContextTypes`와 `getChildContext`를 `MessageList`(context provider)에 추가함으로써, React는 정보를 자동으로 아래로 전달하며 서브트리내의 어떤 컴포넌트든 (여기선 `Button`) `contextTypes`를 정의함으로써 이에 접근할 수 있습니다.
```
Comment (marocchino) [inline docs/docs/12-context.ko-KR.md]: 서브트리내의 -> 하위 트리 내의

File: docs/docs/ref-05-events.ko-KR.md [hunk 9]
```diff
@@ -96,6 +137,7 @@ onFocus onBlur
 DOMEventTarget relatedTarget
 ```
 
+이 포커스 이벤트는 폼 엘리먼트 만이 아닌 모든 React DOM 엘리먼트에서 작동합니다.
```
Comment (marocchino) [inline docs/docs/ref-05-events.ko-KR.md]: 폼 엘리먼트 만이 아닌 -> 폼 엘리먼트뿐만 아니라

File: docs/docs/tutorial.ko-KR.md [hunk 10]
```diff
@@ -26,21 +26,21 @@ next: thinking-in-react-ko-KR.html
 
 ### 서버 구동하기
 
-이 튜토리얼을 시작할 때 필요한 건 아니지만, 나중에 실행 중인 서버에 `POST` 요청을 하는 기능을 추가하게 될 것입니다. 서버를 구성하는 것이 익숙하다면, 본인이 편한 방식대로 서버를 구성해 주세요. 서버사이드에 대한 고민없이 React의 학습 그 자체에 집중하고 싶은 분들을 위해서, 몇 가지 언어로 간단한 서버코드를 작성해 놓았습니다 - JavaScript (Node.js), Python, Ruby, Go, PHP 버전이 있고, GitHub에서 찾아보실 수 있습니다. [소스를 확인](https://github.com/reactjs/react-tutorial/)하거나 [zip 파일을 다운로드](https://github.com/reactjs/react-tutorial/archive/master.zip)하고 시작하세요.
+이 투토리얼을 시작하기 위해, 서버를 구동할 필요가 있습니다. 이 서버는 순수하게 우리가 받고 저장할 데이터의 API 엔드포인트로써만 사용합니다. 이를 가능한한 쉽게하기 위해, 필요한 것만 제공하는 간단한 서버를 몇가지 스크립트 언어로 만들었습니다. **시작하는데 필요한 모든 것이 들어있는 [소스](https://github.com/reactjs/react-tutorial/)를 보시거나 [zip 파일](https://github.com/reactjs/react-tutorial/archive/master.zip)을 다운로드 할 수 있습니다.**
```
Comment (marocchino) [inline docs/docs/tutorial.ko-KR.md]: 투토리얼 -> 튜토리얼

**Comment by arkist** [comment]
LGTM :sunglasses:

**Comment by dalinaum** [comment]
:+1: LGTM

**Comment by jimfb** [comment]
Cool, @marocchino can you squash the commits into a single commit so we can merge?

**Comment by marocchino** [comment]
@jimfb Done. :smile_cat:

**Comment by jimfb** [comment]
Great, thanks @marocchino!

**Comment by spicyj** [comment]
Thank you!

**Comment by arkist** [comment]
Did you really need to have a squash into a single commit? 
We lost individual contribution history. :crying_cat_face:

**Comment by jimfb** [comment]
@arkist Yeah, Paul likes everything to be squashed before merging.  Single commit PRs keep the history a little more linear and makes it easier to revert changes if something goes horribly wrong.

**Comment by spicyj** [comment]
We like to ask for squashed commits so that we don't have lots of "lint fix" or "update based on review comments" commits in the repo but if each commit stands on its own we can take it without squashing. Having commits from different authors definitely seems to fall into that category.

**Comment by arkist** [comment]
@jimfb @spicyj I got it. thanks for explain!

**Pull request merged**